packages/kit/src/core/prerender/prerender.js

```diff
@@ -14,25 +14,17 @@ import { escape_html_attr } from '../../utils/escape.js';
  * @typedef {import('types').Logger} Logger
  */
 
-/** @type {(details: Parameters<PrerenderErrorHandler>[0] ) => string} */
-function format_error({ status, path, referrer, referenceType }) {
-	return `${status} ${path}${referrer ? ` (${referenceType} from ${referrer})` : ''}`;
-}
+/**
+ * @param {Parameters<PrerenderErrorHandler>[0]} details
+ * @param {import('types').ValidatedKitConfig} config
+ */
+function format_error({ status, path, referrer, referenceType }, config) {
+	const message =
+		status === 404 && !path.startsWith(config.paths.base)
+			? `${path} does not begin with \`paths.base\``
```

Comment:
can we include the value? something like:
```suggestion
			? `${path} does not begin with \`paths.base\` value of ${config.kit.paths.base}`
```

Reply:
It could do — I was trying to discourage people from hardcoding it

Reply:
I'm not sure I understand. Hardcoding what?

Reply:
the basepath. if someone sees a message like

> `/some-page` does not begin with `paths.base` value of /foo

they are liable to change the code to

```svelte
<a href="/foo/some-page">some page</a>
```

rather than

```svelte
<script>
  import { base } from '$app/paths';
</script>

<a href="{base}/some-page">some page</a>
```

Reply:
Oh, I see. I was thinking you were talking about hardcoding the base path in the config file, which is of course the only way you'd use it leading me to confusion :smile:  Would this be even more likely to encourage people to do the right thing?
```suggestion
			? `${path} does not begin with \`base\`, which can be imported from \`$app/paths\``
```

Reply:
yeah, i much prefer that